File: csharp/Cola/Cola/Secuencial.cs
namespace Cola
{

  class Secuencial(int dim)
  {
    private readonly int[] items = new int[dim];
    private readonly int dimension = dim;
    private int cantidad = 0;
    private int ultimo = 0;
    private int primero = 0;

    public int Insertar(int item)
    {
      if (cantidad == dimension)
      {
        Console.WriteLine("Cola llena");
        return -1;
      }
      items[ultimo] = item;
      ultimo = (ultimo + 1) % dimension;
      cantidad++;
      return item;
    }

    public int Suprimir()
    {
      if (Vacia())
      {
        Console.WriteLine("Cola vacia.");
        return -1;
      }
      int eliminado = items[primero];
      primero = (primero + 1) % dimension;
      cantidad--;
      return eliminado;
    }

    public void Recorrer()
    {
      for (int i = primero, j = 0; j < cantidad; i = (i + 1) % dimension, j++)
      {
        Console.Write(items[i] + " ");
      }
      Console.WriteLine();
    }

    public bool Vacia()
    {
      return cantidad == 0;
    }
  }
}

File: python/ejercicios/escalones/colas/secuencial.py
import numpy as np


class Cola_Secuencial:
    __cola: np.ndarray
    __primero: int
    __ultimo: int
    __cantidad: int
    __tamaño: int

    def __init__(self, tamaño) -> None:
        self.__tamaño = tamaño
        self.__primero = 0
        self.__ultimo = 0
        self.__cantidad = 0
        self.__cola = np.zeros(self.__tamaño, dtype=object)

    def insertar(self, item):
        if self.llena():
            print("cola llena.")
            return
        self.__cola[self.__ultimo] = item
        self.__ultimo = (self.__ultimo + 1) % self.__tamaño
        self.__cantidad += 1

    def eliminar(self):
        if self.vacia():
            print("pila vacia.")
            return
        temp = self.__cola[self.__primero]
        self.__primero = (self.__primero + 1) % self.__tamaño
        self.__cantidad -= 1
        return temp

    def mostrar(self):
        if self.vacia():
            print("pila vacia.")

        i = self.__primero
        j = 0
        while j < self.__cantidad:
            print(self.__cola[i])
            i = (i + 1) % self.__tamaño
            j += 1

    def vacia(self):
        return self.__cantidad == 0

    def llena(self):
        return self.__cantidad == self.__tamaño

    def obtener_cantidad(self):
        return self.__cantidad


if __name__ == "__main__":
    cola = Cola_Secuencial(3)

    cola.insertar(1)
    cola.insertar(2)
    cola.insertar(3)

    cola.mostrar()
